food-truck-api/src/main/java/food/truck/api/endpoint/TruckEndpoint.java

```
@@ -153,4 +172,54 @@ public String updateRoute(@AuthenticationPrincipal User u, @PathVariable long tr
     public String updateSchedule(@AuthenticationPrincipal User u, @PathVariable long truckId, @RequestBody String data) {
         return ""; // TODO
     }
+
+    @Value
+    private static class PostRouteParams {
+        String routeName;
+        char active;
+    }
+
+    @PostMapping("/truck/{truckId}/create-route")
+    public Route createTruckRoute(@AuthenticationPrincipal User user, @PathVariable long truckId, @RequestBody PostRouteParams data) {
+        Optional<Truck> truck = truckService.findTruckById(truckId);
+        if (truck == null || truck.isEmpty()) {
+            return new Route();
```

Maybe return an error here like above

---

An error would probably be good, though there is the question of whether I've got it set up to expect a response like this (I can't really remember exactly what I wrote at 5:00 this morning).